rct229/rulesets/ashrae9012019/section12/section12rule4.py

```diff
@@ -0,0 +1,159 @@
+            ):
+                if is_space_a_computer_room(rmd_b, space_b["id"]):
+                    for misc_equip_b in space_b["miscellaneous_equipment"]:
+                        if (
+                            misc_equip_b.get("type")
+                            != MISCELLANEOUS_EQUIPMENT.INFORMATION_TECHNOLOGY_EQUIPMENT
+                        ):
+                            return False
+                        else:
+                            return True
+
+        def create_data(self, context, data):
+            rmd_b = context.BASELINE_0
+
+            schedule_b = {
+                mult_sch_b: find_exactly_one_schedule(rmd_b, mult_sch_b)[
+                    "hourly_values"
+                ]
+                for mult_sch_b in find_all(
+                    "$.buildings[*].building_segments[*].zones[*].spaces[*].miscellaneous_equipment[*].multiplier_schedule",
+                    rmd_b,
+                )
+            }
+
+            return {"schedule_b": schedule_b}
+
+        class MiscEquipRule(RuleDefinitionBase):
+            def __init__(self):
+                super(
+                    Section12Rule4.RuleSetModelDescriptionRule.MiscEquipRule,
+                    self,
+                ).__init__(
+                    rmds_used=produce_ruleset_model_description(
+                        USER=False, BASELINE_0=True, PROPOSED=False
+                    ),
+                    required_fields={
+                        "$": [
+                            "multiplier_schedule",
+                        ]
+                    },
+                )
+
+            def get_calc_vals(self, context, data=None):
+                misc_equip_b = context.BASELINE_0
+
+                is_leap_year = data["is_leap_year"]
+                schedule_b = data["schedule_b"]
+
+                if is_leap_year:
+                    DAYS_IN_MONTH[2] = 29
+
+                hourly_multiplier_schedule_b = misc_equip_b["multiplier_schedule"]
+
+                multiplier_schedule_b = schedule_b[hourly_multiplier_schedule_b]
+                expected_hourly_values = []
+                for month in range(1, 13):
+                    expected_hourly_values.extend(
+                        [MONTH_FRACTIONS[month]] * DAYS_IN_MONTH[month] * 24
+                    )
+
+                mask_schedule = [1] * 8784 if is_leap_year else [1] * 8760
+
+                comparison_data = compare_schedules(
+                    multiplier_schedule_b,
+                    expected_hourly_values,
+                    mask_schedule,
+                    is_leap_year,
+                )["total_hours_matched"]
+
+                return {"comparison_data": comparison_data}
+
+            def rule_check(self, context, calc_vals=None, data=None):
+                comparison_data = calc_vals["comparison_data"]
+                is_leap_year = data["is_leap_year"]
+
+                return comparison_data == (8784 if is_leap_year else 8760)
```

Comment:
in here, you just need to match the `total_hours_matched == hours_in_a_year`

Reply:
Addressed.